### PR TITLE
enh(monitoring): add deprecation link in deprecated monitoring pages

### PR DESCRIPTION
**Pull request by adr-mo**
This PR intends to inform the user that the Monitoring listing pages are now deprecated:
```
- Monitoring → Status detail > Service 
- Monitoring → Status detail > Host 
- Monitoring → Status detail > Service detail 
- Monitoring → Status detail > Host detail
```

_Listing pages_
image

_Details pages_
image

When I click the redirection link, regarding where am I:
- `Monitoring → Status detail > Service` : redirects to `/monitoring/resources` page **not filtered**
- `Monitoring → Status detail > Host`: redirects to `/monitoring/resources` page **not filtered**
- `Monitoring → Status detail > Service detail `: redirects to `/monitoring/resources` with the **service detail opened**
- `Monitoring → Status detail > Host detail`: redirects to `/monitoring/resources` with the **host detail opened**

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 19.04.x
- [ ] 19.10.x
- [ ] 20.04.x
- [x] 20.10.x (master)

<h2> How this pull request can be tested ? </h2>

- Go to the following pages
```
- Monitoring → Status detail > Service 
- Monitoring → Status detail > Host 
- Monitoring → Status detail > Service detail 
- Monitoring → Status detail > Host detail
```
- Check that the message is correctly displayed and that the redirection link works

## Checklist

- [x] I followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: www/include/monitoring/objectDetails/hostDetails.php [hunk 1]
```diff
@@ -707,6 +707,12 @@
         $DBRESULT->closeCursor();
         $tpl->assign("isRemote", $isRemote);
 
+        $kernel = \App\Kernel::createForWeb();
+        $resourceController = $kernel->getContainer()->get(
+            \Centreon\Application\Controller\MonitoringResourceController::class
+        );
+
+        $redirection_url = $resourceController->buildHostDetailsUri($host_id);
```
Comment (kduret) [inline www/include/monitoring/objectDetails/hostDetails.php:719]:
```suggestion
        $redirectionUrl = $resourceController->buildHostDetailsUri($host_id);
```

File: www/include/monitoring/objectDetails/hostDetails.php [hunk 2]
```diff
@@ -767,6 +775,14 @@
             "<?php echo $img_en[1];?>"
         );
 
+        function display_deprecated_banner() {
+            var url = "<?php echo $redirection_url; ?>";
```
Comment (kduret) [inline www/include/monitoring/objectDetails/hostDetails.php:783]:
```suggestion
            const url = "<?php echo $redirectionUrl; ?>";
```

File: www/include/monitoring/objectDetails/serviceDetails.php [hunk 3]
```diff
@@ -826,6 +826,13 @@
             $tpl->assign("tools", CentreonUtils::escapeSecure($tools));
         }
 
+        $kernel = \App\Kernel::createForWeb();
+        $resourceController = $kernel->getContainer()->get(
+            \Centreon\Application\Controller\MonitoringResourceController::class
+        );
+
+        $redirection_url = $resourceController->buildServiceDetailsUri($host_id, $service_id);
```
Comment (kduret) [inline www/include/monitoring/objectDetails/serviceDetails.php:838]:
```suggestion
        $redirectionUrl = $resourceController->buildServiceDetailsUri($host_id, $service_id);
```

File: www/include/monitoring/objectDetails/serviceDetails.php [hunk 4]
```diff
@@ -885,6 +894,14 @@
             "<?php echo $img_en[1];?>"
         );
 
+        function display_deprecated_banner() {
+            var url = "<?php echo $redirection_url; ?>";
```
Comment (kduret) [inline www/include/monitoring/objectDetails/serviceDetails.php:902]:
```suggestion
            const url = "<?php echo $redirectionUrl; ?>";
```

File: www/include/monitoring/status/Hosts/host.php [hunk 5]
```diff
@@ -177,6 +177,14 @@
 
 include_once("hostJS.php");
 
+
+$kernel = \App\Kernel::createForWeb();
+$resourceController = $kernel->getContainer()->get(
+    \Centreon\Application\Controller\MonitoringResourceController::class
+);
+
+$redirection_url = $resourceController->buildListingUri([]);
```
Comment (kduret) [inline www/include/monitoring/status/Hosts/host.php:189]:
```suggestion
$redirectionUrl = $resourceController->buildListingUri([]);
```

File: www/include/monitoring/status/Hosts/host.php [hunk 6]
```diff
@@ -392,6 +400,16 @@ function setO(_i) {
     var pending = '<?php echo _("Pending");?>';
     var _keyPrefix;
 
+    display_deprecated_banner();
+
+    function display_deprecated_banner() {
+        var url = "<?php echo $redirection_url; ?>";
```
Comment (kduret) [inline www/include/monitoring/status/Hosts/host.php:409]:
```suggestion
        const url = "<?php echo $redirectionUrl; ?>";
```

File: www/include/monitoring/status/Services/service.php [hunk 7]
```diff
@@ -207,6 +207,13 @@
 include_once("./include/monitoring/status/Common/default_servicegroups.php");
 include_once($svc_path . "/serviceJS.php");
 
+$kernel = \App\Kernel::createForWeb();
+$resourceController = $kernel->getContainer()->get(
+    \Centreon\Application\Controller\MonitoringResourceController::class
+);
+
+$redirection_url = $resourceController->buildListingUri([]);
```
Comment (kduret) [inline www/include/monitoring/status/Services/service.php:219]:
```suggestion
$redirectionUrl = $resourceController->buildListingUri([]);
```

File: www/include/monitoring/status/Services/service.php [hunk 8]
```diff
@@ -486,6 +493,16 @@ function setO(_i) {
     var unknown = '<?php echo _("Unknown");?>';
     var pending = '<?php echo _("Pending");?>';
 
+    display_deprecated_banner();
+
+    function display_deprecated_banner() {
+            var url = "<?php echo $redirection_url; ?>";
```
Comment (kduret) [inline www/include/monitoring/status/Services/service.php:503]:
```suggestion
            const url = "<?php echo $redirectionUrl; ?>";
```

File: www/include/monitoring/objectDetails/hostDetails.php [hunk 9]
```diff
@@ -767,6 +775,14 @@
...
+            jQuery('.pathway').append(
+                '<span style="color:#FF4500;padding-left:10px;font-weight:bold">' +
+                '[Deprecated page. Please use the new <a href="' + url + '">Resource Status</a> page]</span>'
```
Comment (kduret) [inline www/include/monitoring/objectDetails/hostDetails.php]: missing translation

**Review by kduret — DISMISSED** [review]

**Review by jeremyjaouen — DISMISSED** [review]

**Review by callapa — DISMISSED** [review]

**Review by callapa — APPROVED** [review]

**Review by kduret — APPROVED** [review]

**Pull request merged**